biogeochem/EDCanopyStructureMod.F90

```diff
@@ -469,98 +485,150 @@ subroutine DemoteFromLayer(currentSite,currentPatch,i_lyr)
+                           call endrun(msg=errMsg(sourcefile, __LINE__))
+                        end if
+                     end if
 
-                        ! Calculate how much exceeded demotion from filled
-                        ! cohorts must be transferred to this cohort
-                        ! Two requirements: 1) the tacked-on demotion can not also
-                        !     exceed this cohort's area.  And, 2) the tacked-on
-                        !     demotion can't exceed the amount left
+                  endif
+                  currentCohort => currentCohort%shorter      
+               enddo
+               
+            else
 
-                        cc_loss = min(remainder_area, &
-                                      remainder_area_hold * currentCohort%excl_weight/sumweights_old, &
-                                      currentCohort%c_area-currentCohort%excl_weight)
 
+               ! Non-trivial case, at least 1 cohort's demotion
+               ! rate would exceed its area, given the trivial scale factor
+               
+               area_res         = 0._r8
+               scale_factor_res = 0._r8
+               currentCohort => currentPatch%tallest
+               do while (associated(currentCohort))  
+                  if(currentCohort%canopy_layer  ==  i_lyr) then 
+                     area_res         = area_res + &
+                                        currentCohort%c_area * currentCohort%excl_weight * &
+                                        scale_factor_min
+                     scale_factor_res = scale_factor_res + &
+                                        currentCohort%c_area * &
+                                        (1._r8 - (currentCohort%excl_weight * scale_factor_min))
+                  endif
+                  currentCohort => currentCohort%shorter      
+               enddo
+               
+               area_res = demote_area - area_res
+               
+               scale_factor_res = area_res / scale_factor_res
 
-                        ! Reduce the remainder_area
-                        remainder_area = remainder_area - cc_loss
+               currentCohort => currentPatch%tallest
+               do while (associated(currentCohort))  
+                  if(currentCohort%canopy_layer  ==  i_lyr) then 
 
-                        ! Update this cohorts exclusion
-                        currentCohort%excl_weight = currentCohort%excl_weight + cc_loss
-                        
-                        ! Update the sum of weights for cohorts where exceedance can
-                        ! still be spread to
-                        if( currentCohort%excl_weight < (currentCohort%c_area-nearzero) ) then
-                           sumweights = sumweights + currentCohort%excl_weight
+                     currentCohort%excl_weight = currentCohort%c_area * &
+                           (currentCohort%excl_weight * scale_factor_min + &
```

Comment:
@rgknox I don't think you need to multiply by the area here since ```scale_factor_min``` doesn't include the area like ```scale_factor``` correct?

Reply:
I assumed that this should match the technical paper exactly in this case, but I'm thinking that you meant to do this since new line 610 assumes the area is incorporated?

Reply:
ok, I'm looking through this now as well and double checking too